emcc.py

```diff
@@ -1565,6 +1565,9 @@ def get_bitcode_file(input_file):
           options.debug_level = 4
           shared.Settings.DEBUG_LEVEL = 4
 
+      if options.debug_level > 3:
+          newargs.append('-fno-discard-value-names')
```

Comment:
We probably only want this for ASMJS so I would add "and if not Settings.WASM". 

Also we use two spaced for indentation.

Otherwise lgtm

Reply:
Sorry I did not test the upstream LLVM-WASM backend. Luckily seems like the automation also caught it.
Will fix this soon.

Reply:
This should work with the LLVM wasm backend too, I think? We backported this option from newer clang, so upstream should have the option.

Reply:
Oh, is the issue that the wasm backend just can't preserve names anyhow? I see then.